packages/_common/src/frame-processor.ts

```diff
@@ -163,6 +164,31 @@ export class FrameProcessor implements FrameProcessorInterface {
     if (!this.active) {
       return {}
     }
+
+    if (this.paused) {
+
+      const audioBuffer = this.audioBuffer
+
+      const speechFrameCount = audioBuffer.reduce((acc, item) => {
+        return acc + +item.isSpeech
+      }, 0)
+
+      this.active = false
+      this.paused = false
+      this.reset()
+
+      const isSpeech = 0
+      const notSpeech = 1 - isSpeech
+      const probs = { notSpeech, isSpeech }
+
+      if (speechFrameCount >= this.options.minSpeechFrames) {
+        const audio = concatArrays(audioBuffer.map((item) => item.frame))
+        return { probs, msg: Message.SpeechEnd, audio }
+      } else {
+        return { probs, msg: Message.VADMisfire }
+      }
```

Comment:
How about moving these into `pause` method instead of introducing `paused` property.

Reply:
Oh you need `probs`🤔 Your approach seems reasonable.

Reply:
Yes, only `process` has access to the frame and return the messages. The others are control functions for the processor.